waspc/src/Wasp/Generator/FileDraft.hs

```diff
@@ -98,12 +99,23 @@ createCopyFileDraftIfExists dstPath srcPath =
         CopyFD._failIfSrcDoesNotExist = False
       }
 
-createCopyDirFileDraft :: Path' (Rel ProjectRootDir) (Dir a) -> Path' Abs (Dir b) -> FileDraft
-createCopyDirFileDraft dstPath srcPath =
+createCopyDirWithOverwriteFileDraft :: Path' (Rel ProjectRootDir) (Dir a) -> Path' Abs (Dir b) -> FileDraft
+createCopyDirWithOverwriteFileDraft dstPath srcPath = createCopyDirFileDraft dstPath srcPath True
+
+createCopyDirWithMergeFileDraft :: Path' (Rel ProjectRootDir) (Dir a) -> Path' Abs (Dir b) -> FileDraft
+createCopyDirWithMergeFileDraft dstPath srcPath = createCopyDirFileDraft dstPath srcPath False
```

Comment:
> One option is to also replace that Bool with data CopyDirFileDraftDstDirStrategy = RemoveExistingDstDir | WriteOverExistingDstDir, and then you wouldn't define two methods here, but instead write it like createCopyDirFileDraft myDstPath mySrcPath RemoveExistingDstDir.

Uuu I like this a lot, I wanted to avoid the boolean option.

Reply:
What you did is better than boolean option for sure! But this might make it a bit cleaner even.